backend/src/utils/object.ts

```diff
@@ -13,3 +13,12 @@ export function deepFreeze(object) {
 
   return Object.freeze(object)
 }
+
+export function getPropValue(sourceObject: any, dotNotationPath: string) {
+  let returnData = sourceObject
+
+  dotNotationPath.split('.').forEach((subPath) => {
+    returnData = returnData[subPath] || undefined
+  })
+  return returnData
+}
```

Comment:
I've played with this some more and it doesn't differentiate between null and undefined. Here's a possible fix which also avoids `any`:
```suggestion
export function getPropValue<T = unknown>(
  source: unknown,
  path: string
): T | undefined {
  const trimmedPath = path.trim()
  if (!trimmedPath) {
    return source as T
  }

  return trimmedPath.split('.').reduce<any>((acc, key) => {
    return acc != null ? acc[key] : undefined
  }, source)
}
```